.github/workflows/build-wheels-platforms.yml

```diff
@@ -0,0 +1,191 @@
+      - name: OS info
+        if: matrix.os != 'windows-latest'
+        run: |
+          echo "Operating System: ${{ runner.os }}"
+          echo "Architecture: $(uname -m)"
+      - name: OS info
+        if: matrix.os == 'windows-latest'
+        run: |
+          echo "Operating System: ${{ runner.os }}"
+          echo "Architecture: $env:PROCESSOR_ARCHITECTURE"
+
+
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+
+      - name: Setup Python
+        # GitHub action for MacOS M1 does not have Python <= 3.10
+        # Skip setting python on ARM because of missing compatibility: https://github.com/actions/setup-python/issues/108
+        if: matrix.os != 'macos-latest-xlarge' && matrix.os != 'linux-armv7-self-hosted' && matrix.os != 'linux-arm64-self-hosted'
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+
+
+      - name: Setup Python - MacOS M1
+        # Temporary solution until Python version for build will be >= 3.10 (GitHub action support)
+        if: matrix.os == 'macos-latest-xlarge'
+        run: |
+          brew install python@3.8
+          # change python symlink called with default command 'python'
+          ln -s -f /opt/homebrew/bin/python3.8 /usr/local/bin/python
+
+
+      - name: Get Python version
+        run: |
+          python --version
+          python -m pip install --upgrade pip
+
+
+      - name: Install build dependencies
+        run: python -m pip install -r build_requirements.txt
+
+
+      - name: Install additional OS dependencies - Ubuntu
+        if: matrix.os == 'ubuntu-latest'
+        run: |
+          # PyGObject needs build dependecies https://pygobject.readthedocs.io/en/latest/getting_started.html
+          sudo apt install libgirepository1.0-dev gcc libcairo2-dev pkg-config python3-dev gir1.2-gtk-4.0 -y
+
+          # dbus-python needs build dependecies
+          sudo apt-get install cmake build-essential libdbus-1-dev libdbus-glib-1-dev -y
+
+
+      - name: Install additional OS dependencies - MacOS
+        if: matrix.os == 'macos-latest' || matrix.os == 'macos-latest-xlarge'
+        run: |
+          # PyGObject needs build dependecies https://pygobject.readthedocs.io/en/latest/getting_started.html
+          brew install pygobject3 gtk4
+
+          # MacOS M1 additional dependencies
+          if [ "${{ matrix.os }}" == "macos-latest-xlarge" ]; then
+            # Probably temporary fix for gdbgui==0.13.2.0 on MacOS M1 -> gevent==1.5.0 can't be build
+            # The newest version of gevent with M1 support is installed and forced to be used for gdbgui
+            # When requirement wheels are build gdbgui==0.13.2.0 is already built
+
+            arch -arm64 python -m pip install 'cython<3'
+            arch -arm64 python -m pip install 'cffi'
+            arch -arm64 python -m pip install 'gevent'
+            arch -arm64 python -m pip install 'gdbgui==0.13.2.0' --no-build-isolation
```

Comment:
It would be great if don't have to hard code these here. Probably `--no-build-isolation` is the hardest part.

Reply:
Could the build dependencies go into `build_requirements.txt`?

Reply:
I am not sure if it will work, because it is MacOS M1 specific issue and `build_requirements.txt` is common for all platforms, I will try it.

Reply:
OK, I have tried this ([workflow1](https://github.com/espressif/idf-python-wheels/actions/runs/7272987742), [workflow2](https://github.com/espressif/idf-python-wheels/actions/runs/7273106410)) with no success.

And according to [documentation even the newest pip doesn't support `--no-build-isolation`](https://pip.pypa.io/en/latest/reference/requirements-file-format/#supported-options) so for now I would keep this solution with comment and as soon as it will be possible to change it we should do it than.

Reply:
Please try to address this in a follow-up PR because one of the main reasons for this refactoring is to avoid this kind of hacks and have these defined in configuration files.